### PR TITLE
chore(waku-store): improve logging and query traceability

### PR DESCRIPTION
**Pull request by LNSD**
* Added some improvements in the waku store protocol request handler to improve history queries traceability (e.g., peer id, request id).
* Removed redundant logging.

### CODE REVIEW

File: waku/v2/protocol/waku_store/protocol.nim [hunk 1]
```diff
@@ -133,35 +130,34 @@ proc findMessages(w: WakuStore, query: HistoryQuery): HistoryResponse {.gcsafe.}
 proc init*(ws: WakuStore, capacity = StoreDefaultCapacity) =
 
   proc handler(conn: Connection, proto: string) {.async.} =
-    var message = await conn.readLp(MaxRpcSize.int)
-    var res = HistoryRPC.init(message)
-    if res.isErr:
-      error "failed to decode rpc"
+    let buf = await conn.readLp(MaxRpcSize.int)
```
Comment (felicio) [inline waku/v2/protocol/waku_store/protocol.nim:133]: Non-blocking: do you think changes in naming like this (`message` > `buf`) should be done in other files too?

image

File: waku/v2/protocol/waku_store/protocol.nim [hunk 2]
```diff
@@ -133,35 +130,34 @@ proc findMessages(w: WakuStore, query: HistoryQuery): HistoryResponse {.gcsafe.}
...
+    if resReq.isErr():
+      error "failed to decode rpc", peerId=conn.peerId
       waku_store_errors.inc(labelValues = [decodeRpcFailure])
       return
 
-    # TODO Print more info here
-    info "received query", rpc=res.value
+    let req = resReq.value
+
+    info "received history query", peerId=conn.peerId, requestId=req.requestId, query=req.query
```
Comment (felicio) [inline waku/v2/protocol/waku_store/protocol.nim:143]: Note: nice touch with `requestId` 👍 . I understand it should allow grouping of the logs when looking up.

**Review by rymnc — APPROVED** [review]
lgtm!

**Review by jm-clius — APPROVED** [review]
LGTM! Feel free to merge and patch immediately to `test` fleet(s) where these insights will help the debugging.

File: waku/v2/protocol/waku_store/protocol.nim (hunk 1 above)
Comment (LNSD) [reply waku/v2/protocol/waku_store/protocol.nim:133]: Yes, i fully agree, we'll do it as we do small refactoring on the different protocol implementations 😁

File: waku/v2/protocol/waku_store/protocol.nim [hunk 3]
```diff
@@ -133,35 +130,34 @@ proc findMessages(w: WakuStore, query: HistoryQuery): HistoryResponse {.gcsafe.}
 proc init*(ws: WakuStore, capacity = StoreDefaultCapacity) =
 
   proc handler(conn: Connection, proto: string) {.async.} =
-    var message = await conn.readLp(MaxRpcSize.int)
-    var res = HistoryRPC.init(message)
-    if res.isErr:
-      error "failed to decode rpc"
+    let buf = await conn.readLp(MaxRpcSize.int)
+
+    let resReq = HistoryRPC.init(buf)
```
[the remark this answers is not in the document]
Comment (LNSD) [reply waku/v2/protocol/waku_store/protocol.nim:135]: I see. Yes, we should improve the naming of the variables for readability. Thanks for your comment 🙂

**Pull request merged**